tests/test_resource.py

```
@@ -200,3 +203,146 @@ def test_proxy_precedence(self, fake_client):
+    def test_resolve_bearer_token_empty_api_key(self):
+        """Test that empty api_key does not resolve a token."""
+        cfg = kubernetes.client.Configuration()
+        cfg.api_key = {}
+        result = _resolve_bearer_token(token=None, client_configuration=cfg)
+        assert result is None
+
+    def test_save_kubeconfig_write_failure(self, tmp_path):
+        kubeconfig_path = str(tmp_path / "kubeconfig")
+
+        with patch("ocp_resources.utils.kubeconfig.os.open", side_effect=OSError("Permission denied")):
+            save_kubeconfig(path=kubeconfig_path, host="https://api.example.com:6443", token="test-token")
```

Ruff (0.15.6)

[error] Possible hardcoded password assigned to argument: "token"

(S106)